### ISSUE TITLE
SC2129: detect loops

### ISSUE THREAD

**Issue by ChillerDragon** [koalaman/shellcheck#2115]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

echo foo > file.txt
echo foo >> file.txt
for x in {1..5}
do
    echo "foo $x" >> file.txt
done
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

SC2129: Consider using { cmd1; cmd2; } >> file instead of individual redirects.
